nuclia/lib/kb.py

```diff
@@ -223,6 +235,33 @@ def patch_tus_upload(self, upload_url: str, data: bytes, offset: int) -> int:
         handle_http_errors(response)
         return int(response.headers.get("Upload-Offset"))
 
+    def logs(self, type: LogType, month: str) -> list[list[str]]:
+        if self.reader_session is None:
+            raise Exception("KB not configured")
+
+        if type != "FEEDBACK":
+            url = ACTIVITY_LOG_URL.format(type=type, month=month)
+            response: httpx.Response = self.reader_session.get(url)
+            handle_http_errors(response)
+            return [row for row in csv.reader(response.iter_lines())]
+        else:
+            feedback_url = f"{self.url}{FEEDBACK_LOG_URL.format(month=month)}"
+            feedback_response: httpx.Response = self.reader_session.get(feedback_url)
+            handle_http_errors(feedback_response)
+            feedbacks = [row for row in csv.reader(feedback_response.iter_lines())]
+            answers = self.logs(type=LogType.CHAT, month=month)
+            results = [[*feedbacks[0], *answers[0][:-1]]]
+            for feedback in feedbacks[1:]:
+                learning_id = feedback[1]
+                matching_answers = [
+                    answer for answer in answers if answer[1] == learning_id
+                ]
+                if len(matching_answers) > 0:
+                    results.append([*feedback, *matching_answers[0][:-1]])
+                else:
+                    results.append(feedback)
```

Comment:
can you add a comment explaining briefly what you are doing here? it's a bit hard to follow what columns you are merging